src/2_branch.js

```diff
@@ -93,42 +116,52 @@ Branch.prototype['logout'] = function(callback) {
  * @param {?function} callback
  */
 Branch.prototype['track'] = function(event, metadata, callback) {
+	callback = callback || function() {};
 	if (!this.initialized) { return callback(utils.message(utils.messages.nonInit)); }
 	if (typeof metadata == 'function') {
 		callback = metadata;
 		metadata = {};
 	}
-	callback = callback || function() {};
+	
 	this.api(resources.track, {
-		"event": event,
-		"metadata": utils.merge({
-			"url": document.URL,
-			"user_agent": navigator.userAgent,
-			"language": navigator.language
+		event: event,
+		app_id: this.app_id,
+		session_id: this.session_id,
+		metadata: utils.merge({
+			url: document.URL,
+			user_agent: navigator.userAgent,
+			language: navigator.language
 		}, {})
-	}, callback);
+	}, callback({}));
 };
 
 /**
+ * @param {string} identity 
+ * @param {?function} callback
  */
 Branch.prototype['identify'] = function(identity, callback) {
```

Comment:
Yes, that makes perfect sense. I thought that seemed strange, I was just afraid of breaking existing integrations of the WebSDK out in the wild, since the current version takes an object. I suppose I could check if it is passed an object, and pull the string out of it to make it backwards compatible.